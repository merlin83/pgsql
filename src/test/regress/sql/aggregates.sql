--
-- AGGREGATES
--
SELECT avg(four) AS avg_1 FROM onek;

SELECT avg(a) AS avg_32 FROM aggtest WHERE a < 100;

SELECT avg(b) AS avg_107_943 FROM aggtest;

SELECT avg(gpa) AS avg_3_4 FROM student;


SELECT sum(four) AS sum_1500 FROM onek;

SELECT sum(a) AS sum_198 FROM aggtest;

SELECT sum(b) AS avg_431_773 FROM aggtest;

SELECT sum(gpa) AS avg_6_8 FROM student;


SELECT max(four) AS max_3 FROM onek;

SELECT max(a) AS max_100 FROM aggtest;

SELECT max(aggtest.b) AS max_324_78 FROM aggtest;

SELECT max(student.gpa) AS max_3_7 FROM student;


SELECT count(four) AS cnt_1000 FROM onek;

SELECT count(DISTINCT four) AS cnt_4 FROM onek;

select ten, count(*), sum(four) from onek group by ten;

select ten, count(four), sum(DISTINCT four) from onek group by ten;


SELECT newavg(four) AS avg_1 FROM onek;

SELECT newsum(four) AS sum_1500 FROM onek;

SELECT newcnt(four) AS cnt_1000 FROM onek;

