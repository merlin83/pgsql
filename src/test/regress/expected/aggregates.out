QUERY: SELECT avg(four) AS avg_1 FROM onek;
avg_1
-----
    1
(1 row)

QUERY: SELECT avg(a) AS avg_32 FROM aggtest WHERE a < 100;
avg_32
------
    32
(1 row)

QUERY: SELECT avg(b) AS avg_107_943 FROM aggtest;
avg_107_943
-----------
    107.943
(1 row)

QUERY: SELECT avg(gpa) AS avg_3_4 FROM student;
avg_3_4
-------
    3.4
(1 row)

QUERY: SELECT sum(four) AS sum_1500 FROM onek;
sum_1500
--------
    1500
(1 row)

QUERY: SELECT sum(a) AS sum_198 FROM aggtest;
sum_198
-------
    198
(1 row)

QUERY: SELECT sum(b) AS avg_431_773 FROM aggtest;
avg_431_773
-----------
    431.773
(1 row)

QUERY: SELECT sum(gpa) AS avg_6_8 FROM student;
avg_6_8
-------
    6.8
(1 row)

QUERY: SELECT max(four) AS max_3 FROM onek;
max_3
-----
    3
(1 row)

QUERY: SELECT max(a) AS max_100 FROM aggtest;
max_100
-------
    100
(1 row)

QUERY: SELECT max(aggtest.b) AS max_324_78 FROM aggtest;
max_324_78
----------
    324.78
(1 row)

QUERY: SELECT max(student.gpa) AS max_3_7 FROM student;
max_3_7
-------
    3.7
(1 row)

QUERY: SELECT count(four) AS cnt_1000 FROM onek;
cnt_1000
--------
    1000
(1 row)

QUERY: SELECT newavg(four) AS avg_1 FROM onek;
avg_1
-----
    1
(1 row)

QUERY: SELECT newsum(four) AS sum_1500 FROM onek;
sum_1500
--------
    1500
(1 row)

QUERY: SELECT newcnt(four) AS cnt_1000 FROM onek;
cnt_1000
--------
    1000
(1 row)

